aepp/flowservice.py

```diff
@@ -543,19 +542,29 @@ def createFlow(
                 "transformations": [],
                 "scheduleParams": {}
             }
-            if kwargs.get("export_mode",None) is not None:## infer dataset export
-                obj["scheduleParams"]["exportMode"] = kwargs.get("export_mode",None)
-                obj["scheduleParams"]["foldernameTemplate"] = "%DESTINATION%_%DATASET_ID%_%DATETIME(YYYYMMdd_HHmmss)%"
-            if kwargs.get("schedule_timeUnit",None) is not None:
-                obj["scheduleParams"]["timeUnit"] = kwargs.get("schedule_timeUnit",None)
+            if schedule_frequency == "once":
+                obj["scheduleParams"] = {
+                    "timeUnit": "day",
+                    "interval": 0,
+                    "exportMode": "DAILY_FULL_EXPORT"
+                }
+            elif schedule_frequency not in ("minute", "hour"): 
+                if kwargs.get("schedule_timeUnit",None) is not None:
+                    obj["scheduleParams"]["timeUnit"] = kwargs.get("schedule_timeUnit",None)
+                if schedule_end_time is not None:
+                    obj["scheduleParams"]["endTime"] = schedule_end_time
+                if schedule_frequency is not None:
+                    obj["scheduleParams"]["frequency"] = schedule_frequency
+                if schedule_interval is not None:
+                    obj["scheduleParams"]["interval"] = str(schedule_interval)
+                if kwargs.get("export_mode", None) is not None: ## infer dataset export
+                    obj["scheduleParams"]["exportMode"] = kwargs.get("export_mode",None)
+            else:
+                raise ValueError("schedule frequency has to be either once, minute or hour")
+
             if schedule_start_time is not None:
                 obj["scheduleParams"]["startTime"] = schedule_start_time
-            if schedule_end_time is not None:
-                obj["scheduleParams"]["endTime"] = schedule_end_time
-            if schedule_frequency is not None:
-                obj["scheduleParams"]["frequency"] = schedule_frequency
-            if schedule_interval is not None:
-                obj["scheduleParams"]["interval"] = str(schedule_interval)
+            obj["scheduleParams"]["foldernameTemplate"] = output_folder_name
```

Comment:
A Flow can be providing other things that dataset, hence the `output_folder_name` ("%DESTINATION%_%DATASET_ID%_%DATETIME(YYYYMMdd_HHmmss)%") may not be needed or provided with dataset value. 
I would do something like:
- Providing the value in the docstring but still place the default to None. 

```py
if output_folder_name is not None:
    obj["scheduleParams"]["foldernameTemplate"] = output_folder_name

```